src/datadog/tracer.cpp

```diff
@@ -346,9 +383,10 @@ Expected<Span> Tracer::extract_span(const DictReader& reader,
     }
   }
 
-  auto& [trace_id, parent_id, origin, trace_tags, sampling_priority,
-         additional_w3c_tracestate, additional_datadog_w3c_tracestate] =
-      extracted_data;
+  // NOTE(@dmehala): defeat the purpose of the struct. remove following line?
```

Comment:
I added this as a convenience of notation below. The `struct` still serves as the return type of `Extractor`.

It's a matter of taste whether to have all of these prefixed by `extracted_data.` below, or to bring them into scope here.

Reply:
I could have been more explicit, my bad. Indeed, as you mentioned, it serves as a return type. However, struct binding here is error-prone. Let say, if someone in the future change the order in the `struct` definition, it can possibly be a silent bug. Switching `trace_id` and `parent_it` for example.

Reply:
Ok, feel free to remove the bindings and use qualified names below.